### ISSUE TITLE
Unable to set "Iosevka Extended" as my editor font

### ISSUE THREAD

**Issue by IlyaBizyaev** [zed-industries/zed#14096]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

In VSCode, I have the following font configuration:

```
"editor.fontFamily": "'Iosevka Extended', 'monospace', monospace, 'Iosevka Extended Fallback'",
```

I tried to copy it to Zed, which suggested:

```
"buffer_font_family": "Iosevka-Extended",
```

But this setting is ignored, and the default font is used instead.

What works is:

```
"buffer_font_family": "Iosevka",
```
— but I don't like it :)

### Environment

Zed: v0.143.6 (Zed)
OS: Linux Wayland opensuse-tumbleweed 20240705
Memory: 62.5 GiB
Architecture: x86_64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by JunkuiZhang** [comment]
Can you try setting `"buffer_font_family": "Iosevka Extended",`?

**Comment by IlyaBizyaev** [comment]
Tried both with a space and with a dash, neither works :(

**Comment by JunkuiZhang** [comment]
Could you give me a link so that I can download this font and reproduce this issue?

**Comment by IlyaBizyaev** [comment]
You can with:
https://github.com/be5invis/Iosevka/releases/download/v30.3.3/SuperTTC-Iosevka-30.3.3.zip

It's the default version of the last release:
https://github.com/be5invis/Iosevka/blob/v30.3.3/doc/PACKAGE-LIST.md

**Comment by JunkuiZhang** [comment]
I tried with VSCode, it seems `Iosevka Extended` and `Iosevka` are the same.


https://github.com/user-attachments/assets/c4603f0d-01fd-4a86-9c9d-af5dda465f5a

**Comment by IlyaBizyaev** [comment]
On the recording, neither is actually applied to your VSCode, it falls back to some default font. Maybe you didn't install the font correctly...

**Comment by JunkuiZhang** [comment]
Still no visual difference:


https://github.com/user-attachments/assets/bc1cd724-2c5b-4a31-bac4-fbf814de450d

**Comment by IlyaBizyaev** [comment]
I'm not sure how to help with that...

Try following installation instructions for your system: 
https://github.com/be5invis/Iosevka?tab=readme-ov-file#installation

You don't have to record it if it doesn't look like on https://typeof.net/Iosevka/

**Comment by JunkuiZhang** [comment]
I see, VSCode needs to restart to read that font.

https://github.com/user-attachments/assets/c60f9d04-3253-412a-a379-0a7af84ce95d

On my case (win11), there is no font named `Iosevka Extended` or `Iosevka-Extended`

![Screenshot 2024-07-21 174643](https://github.com/user-attachments/assets/b2b1b91d-bfc1-447d-ac82-35b16b1ec05b)

**Comment by IlyaBizyaev** [comment]
It is a width variant of the "Iosevka" font (which is what I guess "54" on your screenshot refers to). It matches the "9 x 3 x 2" scheme on https://typeof.net/Iosevka/, so Windows should be able to access all the variants. I wonder why VSCode doesn't in your case...

**Comment by JunkuiZhang** [comment]
![Screenshot 2024-07-21 180704](https://github.com/user-attachments/assets/a9387c0e-a5a3-4e06-8162-8e9d7a4360a4)


I guess these extended things are something like [font stretch](https://doc.servo.org/font_kit/properties/struct.Stretch.html)?

**Comment by IlyaBizyaev** [comment]
I don't know how it's implemented :(

Here's how it looks in my case:

[2024-07-21 12-15-13.webm](https://github.com/user-attachments/assets/01872ae3-929a-4c26-a101-09ac3887aeb5)

**Comment by JunkuiZhang** [comment]
Can you find a font named `Iosevka Extended` in your font manager?

![Screenshot 2024-07-21 174249](https://github.com/user-attachments/assets/f3fe5bb0-b3a3-4116-8ecf-7933d445b2da)

**Comment by IlyaBizyaev** [comment]
Not as a separate font, no, but I can find it among the 54 variants of the “Iosevka” font, referred to as “Expanded” here for some reason:

![Снимок экрана_20240721_132729](https://github.com/user-attachments/assets/02ad1015-accc-4e46-8873-c95800dcd8a8)

That's in the System Settings of KDE Plasma 6.

**Comment by VortexFlickens** [comment]
I am also facing the exact same issue! Iosevka works but Iosevka-Extended and Iosevka-Medium-Extended doesn't

**Comment by JunkuiZhang** [comment]
From the codes, the `buffer_font_family` should be the family name you see in the font manager, such as `Iosevka`, rather than the font face name, such as `Iosevka-Extended` or `Iosevka Extended`.

**Comment by JunkuiZhang** [comment]
> I just tried again and now it seems to work for me, even though the settings linter is complaining about it. But I can tell it's loading the right one because the default font does not have nerd icons:

On `macOS`, you can use font names like `Expanded`, and `CoreText` will parse them into the correct font face for you.

**Comment by unparadise** [comment]
Have the same issue on Ubuntu. Has anyone found a solution to this yet?

**Comment by Qazalbash** [comment]
This `"buffer_font_family": "Iosevka Fixed SS18"` is working on Zed but not this one `"buffer_font_family": "Iosevka Fixed SS18 Extended"`, and later one was working on VSCode.

**Comment by Pagliacii** [comment]
Maybe we need another option to set the font width. Quoting from [be5invis/Iosevka](https://github.com/be5invis/Iosevka) README.

> 2 widths (Normal and Extended)

**Comment by sourcefrog** [comment]
Yes, it looks like there's a distinct "width" field that Zed does not yet have a way to configure. 

[In CSS and in font-kit this is called "stretch"](https://drafts.csswg.org/css-fonts-3/#font-stretch-prop).

It looks like some amount of support for this already exists in the lower-level libraries e.g. https://docs.rs/font-kit/latest/font_kit/properties/struct.Stretch.html but it's not hooked all the way through gpui and to a setting.

**Comment by H-E3** [comment]
font setting “Iosevka Extended” also does not work on Obsidian.

**Comment by lakshits11** [comment]
same issue observed on Zed 0.233.8

**Comment by pointydev** [comment]
Unable to set via UI, but can force `"Iosevka Extended"` in my settings file (despite Zed complaining it is invalid when saving).

![](https://github.com/user-attachments/assets/ee4f107a-750c-450f-a5a4-52a7665909c2)

- [Iosevka 34.4.0](https://copr.fedorainfracloud.org/coprs/peterwu/iosevka/build/10346976/)
- Zed Nightly [`1.2.0+nightly` (from Terra)](https://github.com/terrapkg/packages/actions/runs/25356109755/job/74345647490)

**Comment by Vivraan** [comment]
Still doesn't work on Zed's stable release branch. I use Iosevka's SS05 variant, so the naming is a mystery to me.

**Comment by wrongcitizen** [comment]
This bug is still present and can be reproduced on the latest stable Zed.

Unable to set buffer font to "Iosevka Extended", even though the font is installed and works in all other editors on the system.

<img width="976" height="144" alt="Image" src="https://github.com/user-attachments/assets/86a73faa-9b2e-4e5e-9977-d35ee6f3c77f" />

<img width="1692" height="262" alt="Image" src="https://github.com/user-attachments/assets/37cff9be-5967-4187-8287-7e0b66b02b94" />

Environment:
Zed: 1.19.2
OS: Arch Linux
DE: KDE Plasma 6.7.5
